### PR TITLE
Updates PyYAML to 5.4.1

### PR DESCRIPTION
**Pull request by emkll**
Closes https://github.com/freedomofpress/securedrop-debian-packaging/issues/219

This should be reviewed/merged in conjunction with the changes in https://github.com/freedomofpress/securedrop-debian-packaging/pull/220 (The build CI target will fail until that PR is merged)

- [x] review/merge https://github.com/freedomofpress/securedrop-debian-packaging/pull/220
- [x] restart CI
- [x] CI should pass, this PR should be reviewed/merged

Based on the description of CVE-2020-14343:
> A vulnerability was discovered in the PyYAML library in versions before 5.4, where it is susceptible to arbitrary code execution when it processes untrusted YAML files through the full_load method or with the FullLoader loader. Applications that use the library to process untrusted input may be vulnerable to this flaw. This flaw allows an attacker to execute arbitrary code on the system by abusing the python/object/new constructor. This flaw is due to an incomplete fix for CVE-2020-1747. See CVE-2020-14343.

We don't use `full_load` of the `FullLoader` (we only use it to load the configuration via `safe_load`) in production, so we shouldn't be vulnerable here. While we should update the version to match the ones use in dev, the changes here will not warrant a proxy release (we can release these updated wheels as part of a future release)

### CODE REVIEW

**Comment by creviera** [comment]
kicking ci

File: .circleci/config.yml [hunk 1]
```diff
@@ -77,3 +77,15 @@ workflows:
     jobs:
       - test-buster
       - build-buster
+
+  nightly:
```
Comment (creviera) [inline .circleci/config.yml:81]: b-e-a-utiful

File: build-requirements.txt [hunk 2]
```diff
@@ -3,7 +3,7 @@ chardet==3.0.4 --hash=sha256:e5cf39014befb85add77118fdc946f0a3387df7840235ba5d20
 furl==2.0.0 --hash=sha256:9f50360f6e4a0f1d0a35fb4997878e7186a73331f0fde5f6fc9b1bb9f006e6cc
 idna==2.7 --hash=sha256:a967a752dba5492be34f30f8633cee512d2ec923fe3a2e201d87353b540d2955
 orderedmultidict==1.0 --hash=sha256:f6022beda2b3387c61e6eb7e0e1e3e2832fd9f55f3f64d4b4b226eea7487327f
-pyyaml==5.3.1 --hash=sha256:ff021f812a5dbaeb10678462952bb447bf6b322ab54923f5591811056a827c15
+pyyaml==5.4.1 --hash=sha256:c2ab2737721fe6f5545fa830312fe7b77e74932c8bbd780a17a3d7df2ff65f99
```
Comment (creviera) [inline build-requirements.txt:6]: :heavy_check_mark:

File: dev-requirements.txt [hunk 3]
```diff
@@ -130,41 +145,59 @@ mypy==0.761 \
...
-    --hash=sha256:95f71d2af0ff4227885f7a6605c37fd53d3a106fcab511b8860ecca9fcf400ee \
-    --hash=sha256:b8eac752c5e14d3eca0e6dd9199cd627518cb5ec06add0de9d32baeee6fe645d \
-    --hash=sha256:cc8955cfbfc7a115fa81d85284ee61147059a753344bc51098f3ccd69b0d7e0c \
-    --hash=sha256:d13155f591e6fcc1ec3b30685d50bf0711574e2c0dfffd7644babf8b5102ca1a \
-    # via -r requirements.in, vcrpy
+    --hash=sha256:f661252913bc1dbe7fcfcbf0af0db3f42ab65aabd1a6ca68fe5d466bace94dae
+    # via
+    #   -r dev-requirements.in
+    #   flake8
+pyyaml==5.4.1 \
```
Comment (creviera) [inline dev-requirements.txt:176]: looks like you were able to catch em all: https://pypi.org/project/PyYAML/#files

**Review by creviera — APPROVED** [review]
lgtm - can be merged once ci passes

**Comment by creviera** [comment]
CI failing:

```
Missing sha256sum for package: #
make: *** [Makefile:5: securedrop-proxy] Error 1
```

**Comment by emkll** [comment]
Thanks @creviera , I've reverted to the old pip-tools format for the requirements file to unblock this. I've opened https://github.com/freedomofpress/securedrop-debian-packaging/issues/225 to track improvements to the build scripts, which we should tackle in the near future (as we may need to revisit this logic as part of https://github.com/freedomofpress/securedrop-debian-packaging/issues/218)

This should now be ready for re-review.

**Review by kushaldas — APPROVED** [review]
All points are good, matched hash. Approved.

**Pull request merged**